### PR TITLE
[plugin] extract flowResourcesFolder into AdapterBase

### LINKED ISSUE #10318: Give BuildUtil possability to  define FLOW_RESOURCES_FOLDER
Will do a PR

Extract  FrontendUtils.DEFAULT_FLOW_RESOURCES_FOLDER into Adapter to have more possibilitiea to control where the resource should be placed. For example, in not maven Layouts, the base path '/target' is not the best idea

https://github.com/vaadin/flow/blob/0be0cb985311bfe9a95eb5b264459a42d6b24ca9/flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/plugin/base/BuildFrontendUtil.java#L133

### PR DESCRIPTION
**Pull request by stbischof**
Gives Plugins possability to define FLOW_RESOURCES_FOLDER. 

Closes #10318

### CODE REVIEW

File: flow-plugins/flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/FlowModeAbstractMojo.java [hunk 1]
```diff
@@ -372,4 +373,10 @@ public File webpackOutputDirectory() {
     public boolean isJarProject() {
         return "jar".equals(project.getPackaging());
     }
+
+    @Override
+    public String flowResourcesFolder() {
+
```
Comment (caalador) [inline flow-plugins/flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/FlowModeAbstractMojo.java:379]:
```suggestion
```

File: flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/plugin/base/PluginAdapterBase.java [hunk 2]
```diff
@@ -274,5 +274,13 @@ default boolean isUseDeprecatedV14Bootstrapping() {
      */
     File webpackOutputDirectory();
 
+    /**
+     *  Default folder for copying front-end resources present in the classpath
+     * jars.
+     * 
+     * @return {@link String}
+     */
```
Comment (caalador) [inline flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/plugin/base/PluginAdapterBase.java:282]:
```suggestion
    /**
     * Default target folder to copy front-end resources to from jars present
     * on the classpath and project local frontend files in
     * frontendResourcesDirectory.
     *
     * @return {@link String}
     */
```

**Review by caalador — CHANGES_REQUESTED** [review]
`BuildFrontendUtil::runNodeUpdater` L241 still has the hardcoded folder  
```
        File flowResourcesFolder = new File(adapter.npmFolder(),
                DEFAULT_FLOW_RESOURCES_FOLDER);
```

Also as this is only set in BuildFrontend I guess running only prepare will generate a different result as it will still use the default target directory `File flowResourcesFolder = new File(baseDir, DEFAULT_FLOW_RESOURCES_FOLDER);` `DevModeInitialize:L278` The given folder should be also delegated here when only running PrepareFrontend

**Comment by stbischof** [comment]
There is an other issue that will fix the static use of the target folder.

**Comment by caalador** [comment]
But that is another case. Here we are adding the `flowResourcesFolder` through the plugin which should probably be written to the token file to get it into the configuration object in prepare-frontend and then removed from there in build-frontend step as then we shouldn't have any paths stored in the token file.

DevModeInitializer should then read the `PARAM_FLOW_RESOURCES` from the configuration as `File flowResourcesFolder = new File(baseDir, config.getStringProperty(PARAM_FLOW_RESOURCES, DEFAULT_FLOW_RESOURCES_FOLDER));`

**Comment by caalador** [comment]
Is there still a need for this change as #10335 was fixed and the resources will go according to the defined build directory?

**Comment by stbischof** [comment]
yes this should be enough!

**Pull request closed without merging**